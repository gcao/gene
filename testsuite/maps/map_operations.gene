#!/usr/bin/env gene run
# TEST: Map/dictionary operations

# Map creation
(var person {
  "name" "Alice"
  "age" 30
  "city" "New York"
})
(print "person: " person)

# Map access
(print "name: " (person "name"))
(print "age: " (person "age"))

# Map with symbol keys
(var config {
  :host "localhost"
  :port 8080
  :debug true
})
(print "host: " (config :host))
(print "port: " (config :port))

# Map methods
(print "keys: " (.keys person))
(print "values: " (.values person))
(print "has name: " (.has_key person "name"))
(print "has email: " (.has_key person "email"))

# Map update
(.set person "email" "alice@example.com")
(print "after adding email: " person)

# Nested maps
(var data {
  "user" {
    "id" 123
    "profile" {
      "bio" "Software developer"
    }
  }
})
(print "nested access: " ((data "user") "id"))
(print "deep nested: " (((data "user") "profile") "bio"))