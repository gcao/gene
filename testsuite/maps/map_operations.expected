person: {"name" "Alice" "age" 30 "city" "New York"}
name: Alice
age: 30
host: localhost
port: 8080
keys: ["name" "age" "city"]
values: ["Alice" 30 "New York"]
has name: true
has email: false
after adding email: {"name" "Alice" "age" 30 "city" "New York" "email" "alice@example.com"}
nested access: 123
deep nested: Software developer